src/gflownet/models/graph_transformer.py

```diff
@@ -104,12 +109,19 @@ def forward(self, g: gd.Batch, cond: torch.Tensor):
         for i in range(self.num_layers):
             # Run the graph transformer forward
             gen, trans, linear, norm1, ff, norm2, cscale = self.graph2emb[i * 7:(i + 1) * 7]
-            agg = gen(o, aug_edge_index, aug_e)
-            l_h = linear(trans(torch.cat([o, agg], 1), aug_edge_index, aug_e))
             cs = cscale(c[aug_batch])
-            scale, shift = cs[:, :l_h.shape[1]], cs[:, l_h.shape[1]:]
-            o = norm1(o + l_h * scale + shift, aug_batch)
-            o = norm2(o + ff(o), aug_batch)
+            if self.ln_type == 'post':
+                agg = gen(o, aug_edge_index, aug_e)
+                l_h = linear(trans(torch.cat([o, agg], 1), aug_edge_index, aug_e))
+                scale, shift = cs[:, :l_h.shape[1]], cs[:, l_h.shape[1]:]
+                o = norm1(o + l_h * scale + shift, aug_batch)
+                o = norm2(o + ff(o), aug_batch)
+            else:
+                agg = gen(norm1(o, aug_batch), aug_edge_index, aug_e)
+                l_h = linear(trans(torch.cat([o, agg], 1), aug_edge_index, aug_e))
+                scale, shift = cs[:, :l_h.shape[1]], cs[:, l_h.shape[1]:]
+                o = o + l_h * scale + shift
+                o = o + ff(norm2(o, aug_batch))
```

Comment:
Why is `norm2` before `ff` when `ln_type == "pre"`?

Reply:
the `pre`-LayerNorm normalizes the inputs of both modules (which are the transformer bit `trans`, and the MLP bit `ff`) instead of their outputs:
![image](https://user-images.githubusercontent.com/654594/217579520-d9fd1b0b-5b2a-42d6-8605-5385d08e3c62.png)

Reply:
Actually, you made me spot an inconsistency. I was norming the input of `gen` but not `trans`! Fixed in latest commit.

Reply:
Meant to do that 💪